xdsl/dialects/math.py

```diff
@@ -529,12 +462,18 @@ class FPowIOp(IRDLOperation):
     """
 
     name = "math.fpowi"
+
+    T1: ClassVar = VarConstraint("T1", floatingPointLike)
+    T2: ClassVar = VarConstraint("T2", signlessIntegerLike)
```

Comment:
seems unnecessary, although I guess it doesn't hurt much

Reply:
I agree, there's no reason to have this variable

Reply:
I'll amend, it's a leftover from reusing the base classes.

Reply:
It seems like the assembly format cannot infer the type of `result`

>  type of result 'result' cannot be inferred, consider adding a 'type($result)' directive to the custom assembly format

Hence, I ended up keep one of them.